components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java

```diff
@@ -135,170 +135,179 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
 
         synchronized ((consumerKey + ":" + authorizedUser + ":" + scope).intern()) {
 
+            AccessTokenDO existingAccessTokenDO = null;
             // check if valid access token exists in cache
             if (cacheEnabled) {
-                AccessTokenDO accessTokenDO = (AccessTokenDO) OAuthCache.getInstance().getValueFromCache(cacheKey);
-                if (accessTokenDO != null) {
+                existingAccessTokenDO = (AccessTokenDO) OAuthCache.getInstance().getValueFromCache(cacheKey);
+                if (existingAccessTokenDO != null) {
                     if (log.isDebugEnabled()) {
-                        log.debug("Retrieved active Access Token" +
-                                " for Client Id : " + consumerKey + ", User ID :" + authorizedUser +
-                                " and Scope : " + scope + " from cache");
+                        log.debug("Retrieved active Access Token for Client Id : " + consumerKey + ", User ID :"
+                                + authorizedUser + " and Scope : " + scope + " from cache");
                     }
 
-                    long expireTime = OAuth2Util.getTokenExpireTimeMillis(accessTokenDO);
+                    long expireTime = OAuth2Util.getTokenExpireTimeMillis(existingAccessTokenDO);
+
+                    if ((expireTime > 0 || expireTime < 0)) {
```

Comment:
can't we have this as expireTime !=0

Reply:
This class is deprecated and I did minimal changes to get the respective issue fixed rather than existing logic.
This logic was improved in the ResposeHandlerUtil class which will be used going forward.